evm_arithmetization/src/cpu/kernel/asm/beacon_roots.asm

```diff
@@ -46,3 +49,34 @@ after_beacon_roots_storage_insert:
     // stack: account_storage_root_ptr_ptr, new_storage_root_ptr, retdest
     %mstore_trie_data
     JUMP
+
+delete_root_idx_slot:
+    // stack: root_idx, 0, write_beacon_roots_to_storage, timestamp_idx, timestamp, retdest
+    PUSH after_root_idx_slot_delete
+    SWAP2 POP
+    // stack: root_idx, after_root_idx_slot_delete, write_beacon_roots_to_storage, timestamp_idx, timestamp, retdest
+    %slot_to_storage_key
+    // stack: storage_key, after_root_idx_slot_delete, write_beacon_roots_to_storage, timestamp_idx, timestamp, retdest
+    PUSH 64 // storage_key has 64 nibbles
+    %get_storage_trie(@BEACON_ROOTS_ADDRESS)
+    // stack: storage_root_ptr, 64, storage_key, after_root_idx_slot_delete, write_beacon_roots_to_storage, timestamp_idx, timestamp, retdest
+    DUP1 ISZERO %jumpi(skip_empty_slot)
+    // stack: storage_root_ptr, 64, storage_key, after_root_idx_slot_delete, write_beacon_roots_to_storage, timestamp_idx, timestamp, retdest
+    %jump(mpt_delete)
```

Comment:
I think that `mpt_delete` assumes that the key is present in the trie, otherwise it might be buggy. So should we check this first?

Reply:
Yeah it would panic, but I have already added the check above:
`DUP1 ISZERO %jumpi(skip_empty_slot)` as we would get a 0 pointer only in case we didn't find it, unless I am mistaken?

Reply:
Ah yeah you're right. I've added an `mpt_read` check.